### ISSUE TITLE
Does not work with gulp v4

### ISSUE THREAD

**Issue by fracz** [Mikhus/gulp-help-doc#1]
Sadly, this does not work with gulp v4. Given the following gulpfile, it shows an empty output from `gulp help`.

``` js
var gulp = require('gulp');

/**
 * Something
 * @task {sth}
 */
gulp.task(function sth(done) {
    done();
});

gulp.task(function help(done) {
    require('gulp-help-doc')(gulp);
    done();
});
```

Do you suspect why it does not work? Would it be hard to fix?

Thank you!

**Comment by Mikhus** [comment]
Thanks, I will investigate this ASAP

**Comment by Mikhus** [comment]
I see no problem with gulp4.

First of all, gulp v4 is unstable and is not released yet, but nevermind(I cloned their 4.0 branch for test).
Second, the code you provided is incorrect as far as gulp task requires it's name as first argument.
And third, when the code is correct everything works as expected.

In your case it would be

``` javascript
/**
 * Something
 * @task {sth}
 */
gulp.task('sth', function sth(done) {
    done();
});

/**
 * Prints this info.
 * @task {help}
 */
gulp.task('help', function help(done) {
    require('gulp-help-doc')(gulp);
    done();
});
```

**Comment by fracz** [comment]
Thank you for your respone.

However, the [Changelog](https://github.com/gulpjs/gulp/blob/4.0/CHANGELOG.md#task-system-changes) says:

> added single argument syntax for gulp.task which allows a named function to be used as the name of the task and task function.

So the syntax I had posted is valid and works for me.

I suppose you have gulp 3 installed globally and that is the reason of not understanding the new syntax. Try

```
npm rm -g gulp
npm install -g gulp-cli
```

(only the cli should be installed globally).

**Comment by Mikhus** [comment]
I don't have any modules installed globally. I have my PATH refering local ./node_modules/.bin, so I never install globally anything.
Gulp v4 is not released, and the thing I cloned from their v4 branch throws an error:

```
Error: Task requires a name that is a string
    at Gulp.Orchestrator.add (/home/mikhus/Projects/test/node_modules/gulp/node_modules/orchestrator/index.js:41:10)
    at Object.<anonymous> (/home/mikhus/Projects/test/gulpfile.js:22:6)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Liftoff.handleArguments (/home/mikhus/Projects/test/node_modules/gulp/bin/gulp.js:116:3)
```

As you may see from a stack trace it is not global reference.

All kind regards,
Mike

**Comment by fracz** [comment]
It's weird then. Can you please try installing the gulp v4 with `npm install 'gulpjs/gulp.git#4.0' --save-dev`?

**Comment by Mikhus** [comment]
OK

**Comment by Mikhus** [comment]
My fault! I cloned but forgot to switch to a proper branch, so it was actually 3.9 :)
Taking this into work, thank you for contribution!

**Comment by Mikhus** [comment]
Fixed in release v1.0.5
